test/NuGet.Core.Tests/NuGet.Commands.Test/MSBuildRestoreUtilityTests.cs

```diff
@@ -2655,6 +2655,377 @@ public void MSBuildRestoreUtility_AddPackageDownloads_InvalidVersionRange(string
             var exception = Assert.Throws<ArgumentException>(() => MSBuildRestoreUtility.AddPackageDownloads(spec, msbuildItems));
         }
 
+        [Fact]
+        public void MSBuildRestoreUtility_GetDependencySpec_CentralVersionIsMergedWhenCPVMEnabled()
+        {
+            var projectName = "acpvm";
+            using (var workingDir = TestDirectory.Create())
+            {
+                // Arrange
+                var projectUniqueName = "482C20DE-DFF9-4BD0-B90A-BD3201AA351A";
+                var project1Root = Path.Combine(workingDir, projectName);
+                var project1Path = Path.Combine(project1Root, $"{projectName}.csproj");
+
+                var items = new List<IDictionary<string, string>>();
+
+                items.Add(new Dictionary<string, string>()
+                {
+                    { "Type", "ProjectSpec" },
+                    { "ProjectName", projectName },
+                    { "ProjectStyle", "PackageReference" },
+                    { "ProjectUniqueName", projectUniqueName },
+                    { "ProjectPath", project1Path },
+                    { "TargetFrameworks", "netcoreapp3.0" },
+                    { "CrossTargeting", "true" },
+                    { "CentralPackageVersionsEnabled", "true"}
+                });
+
+                // Package reference
+                items.Add(new Dictionary<string, string>()
+                {
+                    { "Type", "Dependency" },
+                    { "ProjectUniqueName", projectUniqueName },
+                    { "Id", "x" },
+                    { "TargetFrameworks", "netcoreapp3.0" },
+                    { "IncludeAssets", "build;compile" },
+                    { "CrossTargeting", "true" },
+                });
+
+                // Package reference with version 
+                items.Add(new Dictionary<string, string>()
+                {
+                    { "Type", "Dependency" },
+                    { "ProjectUniqueName", projectUniqueName },
+                    { "Id", "y" },
+                    { "VersionRange", "1.2.1" },
+                    { "IsImplicitlyDefined", "true" },
```

Comment:
This conflicts with the other IsImplicitlyDefined/AutoReferenced check in RestoreCommand https://github.com/NuGet/NuGet.Client/pull/3198/files#diff-ae3ea71e74915c8f0634a376599d6b87R406

Reply:
The test does not test the behavior in the RestoreCommand but the GetDependencySpec.

Reply:
I understand that, but tests should ideally replicate happy scenarios as close to the real ones as possible. 

Don't want to block it here though, and it has a very low relevance.